scripts/chatty_news.js

```diff
@@ -0,0 +1,512 @@
+                                } else {
+                                    var parser = new DOMParser();
+                                    //replace handles badly (nonexistant) html encoded article titles
+                                    var doc = parser.parseFromString(chattynews.html_entity_decode(request.responseText),"text/xml"); //.replace(/& /g,"&amp;")
+                                }
+                                callbackDoc(doc.documentElement);
+                            }
+                            if(callbackText!=null){
+                                callbackText(request.responseText);
+                            }
+                        }
+                    }
+                    if (request.readyState == 4 && request.status != 200) {
+                        console.log("Requested '"+url+"' bur received status "+request.status+"  text:"+request.responseText);
+                        if(callbackDoc!=null){
+                            callbackDoc(null);
+                        }
+                        if(callbackText!=null){
+                            callbackText(null);
+                        }
+                    }
+                }
+             
+                request.send(null);
+            },
+            parseRSS: function(data){
+                var count = Number(getSetting("chatty_news_article_count"));
+                if(count < 1){ 
+                    count = 7;
+                }
+                var items = data.getElementsByTagName('item');
+                
+                var output = '<h3>NEWS HEADLINES</h3><ul>';
+                for (var i = 0; i < ((items.length<count)?items.length:count); ++i) {
+                    var title = chattynews.valueFromTagName(items[i], 'title');
+                    var link = chattynews.valueFromTagName(items[i], 'link');
+                    var desc = chattynews.valueFromTagName(items[i], 'description').replace(/<[^>]*>/g,'').replace(/"/g,'&quot;');
+                    output += '<li><a href ="' + link + '" title="'+desc+'" target="_blank">' + title + '</li>\n';
+                }
+                output += '</ul>';
+                var introwrap = document.getElementById('main');//'chatty_intro_wrap'
+                if(introwrap!=null){
+                    var RSSOutput = document.createElement("div");
+                    RSSOutput.setAttribute('id','SNCNheadlines');
+                    RSSOutput.innerHTML = output;
+                    
+                    var content = introwrap.getElementsByClassName("content")[0];
+                    introwrap.insertBefore(RSSOutput,content);
+                }
+                
+                if(getSetting("chatty_news_article_news")){
+                    var elms = document.evaluate("//div[contains(@class,'fpauthor_14475')]/div[contains(@class,'postbody')]/a[contains(@href,'article/')]", document, null, XPathResult.UNORDERED_NODE_SNAPSHOT_TYPE, null);
```

Comment:
Don't use `document.evaluate()` outside of the root event handler if it can be avoided. It's not as readable or easy to debug as `querySelector()`.

Reply:
This statement can easily be replaced with a `[ ...document.querySelectorAll(...) ]` call (spread the NodeList into an Array to iterate with `for .. of`). That would be significantly more readable and testable than using XPath. In the near future (once the ES6 fork is done) the remaining XPath call (in `chromeshack_posts.js`) will go away and be replaced with `querySelectorAll(...)`.

It's up to you if you want to fix this or let me do it once I merge this. Maybe one last pass of refactoring to polish things up?